sources/packages/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -273,4 +287,124 @@ export class AssessmentControllerService {
+          ApplicationExceptionStatus.Pending,
+          ApplicationExceptionStatus.Declined,
+        );
+
+      if (applicationExceptions.length > 0) {
+        const applicationExceptionArray: RequestAssessmentSummaryAPIOutDTO[] =
+          applicationExceptions.map((applicationException) => ({
+            id: applicationException.id,
+            submittedDate: applicationException.createdAt,
+            status: applicationException.exceptionStatus,
+            requestType: RequestAssessmentTypeAPIOutDTO.StudentException,
+          }));
+        return requestAssessmentSummary.concat(applicationExceptionArray);
+      }
+    }
+
+    const studentAppeal =
+      await this.studentAppealService.getPendingAndDeniedAppeals(
+        applicationId,
+        studentId,
+      );
+    const studentAppealArray: RequestAssessmentSummaryAPIOutDTO[] =
+      studentAppeal.map((appeals) => ({
+        id: appeals.id,
+        submittedDate: appeals.submittedDate,
+        status: appeals.status,
+        requestType: RequestAssessmentTypeAPIOutDTO.StudentAppeal,
+      }));
+    return requestAssessmentSummary.concat(studentAppealArray);
+  }
+
+  /**
+   * Method to get history of assessments for an application,
+   * i.e, this will have original assessment for the
+   * student application, and all approved student
+   * appeal and scholastic standings for the application
+   * which will have different assessment status.
+   * @param applicationId, application number.
+   * @param studentId applicant student.
+   * @returns summary of the assessment history for a student application.
+   */
+  async getAssessmentHistorySummary(
+    applicationId: number,
+    studentId?: number,
+  ): Promise<AssessmentHistorySummaryAPIOutDTO[]> {
+    const [assessments, unsuccessfulScholasticStanding] = await Promise.all([
+      this.assessmentService.assessmentHistorySummary(applicationId, studentId),
+      this.studentScholasticStandingsService.getUnsuccessfulScholasticStandings(
+        applicationId,
+      ),
+    ]);
+    const history: AssessmentHistorySummaryAPIOutDTO[] = assessments.map(
+      (assessment) => ({
+        assessmentId: assessment.id,
+        submittedDate: assessment.submittedDate,
+        triggerType: assessment.triggerType,
+        assessmentDate: assessment.assessmentDate,
+        status: assessment.status,
+        offeringId: assessment.offering.id,
+        programId: assessment.offering.educationProgram.id,
+        studentAppealId: assessment.studentAppeal?.id,
+        applicationExceptionId: assessment.application.applicationException?.id,
+        studentScholasticStandingId: assessment.studentScholasticStanding?.id,
+      }),
+    );
+    // Add unsuccessful scholastic standing to the top of the list, if present.
+    // For unsuccessful scholastic standing, status is always "completed" and
+    // "createdAt" is "submittedDate".
+    if (unsuccessfulScholasticStanding) {
+      history.unshift({
```

Comment:
For my understanding, was unshift used here to ensure the most recent record is displayed first ?

Reply:
It was used to add elements in the beginning of the array and this logic was existing already.